lightning/src/util/config.rs

```diff
@@ -246,6 +246,10 @@ pub struct ChannelConfig {
 	/// [`Normal`]: crate::chain::chaininterface::ConfirmationTarget::Normal
 	/// [`Background`]: crate::chain::chaininterface::ConfirmationTarget::Background
 	pub force_close_avoidance_max_fee_satoshis: u64,
+	/// If this is set to false, we do not auto-close channel with pending time-sensitive outputs,
+	/// of which the feerate has been detected as stalling.
+	/// Default value: true.
+	pub should_autoclose: bool,
```

Comment:
Maybe we should make this some kind of knob to configure how aggressively we do this instead? eg right now we require a feerate increase of 30%, maybe we should make that a knob and users who want to turn off autoclose can set it to `u64::max_value()`?

Reply:
What feerate we would like the commitment transaction to be committed with ? Ideally we would have visibility on the current mempool feerate groups composition to predict a worst-case evolution of those ones based on historical data. Assuming this worst-case, we should guarantee our feerate stay in-sync with the top feerate group for block inclusion during our `CLTV_CLAIM_BUFFER` (once we unilateraly break the chain we can't update it anymore). The current "30%" should represent our rate of fidelity of the worst-case, if a node operator would like to diverge from the worst-case, they should tweak it accordingly.                                                                                  

Even further, to have a more complete model, we should also tweak AUTOCLOSE_TIMEOUT to cover the velocity of the worst-case scenario ("Once we detected evolution towards unsafe mempool feerate groups,          
how fast we can react to correct the included feerate ?"). The issue being there, our reaction time is dependent on our counterparty interactivity latency, and there we could demand input from the node operator ("what's the liveliness reliability of this peer ?") to adjust in consequence AUTOCLOSE_TIMEOUT. The current AUTOCLOSE_TIMEOUT should be block-long enough to let our fee-estimator and our                    counterparty converge, but also short enough to match worst-case congestion velocity.

Ultimately the limitating safety factor is our counterparty latency. And that one can be quite extreme if our counterparty isn't robust against low-odds, high-impact risks (data center fire, DNS server outage, etc). Assuming package relay support and moving to anchor output as a fee-bumping mechanism remove this safety dependency on counterparty latency.

Or if you have a more rule-of-thumb thinking on wat to advice node operators to configure this potential knob ?

Reply:
I don't think there is a clear answer, sadly. Indeed, no fee estimator (that I'm aware of) is really tuned for the lightning threat model. That said, the answer is likely some function over the trust a user has for their counterparty, which leads me to believe it should be a knob. If you're running a node that opens channels automatically with any counterparty, you probably want something pretty tight (30% seems reasonable?), if you're a node that only makes outbound channel opens with trusted peers, maybe 500% is more reasonable.